resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -8,19 +8,34 @@ jruby-puppet: {
     # used by the `puppetserver gem` command line tool.
     gem-home: /opt/puppetlabs/server/data/puppetserver/jruby-gems
 
+
+    # PLEASE NOTE: Use caution when modifying the below settings. Modifying
+    # these settings will change the value of the corresponding Puppet settings
+    # for Puppet Server, but not for the puppet CLI tools. This likely will not
+    # be a problem with master-var-dir, master-run-dir, or master-log-dir unless
+    # some critical setting in puppet.conf is interpolating the value of one
+    # of the corresponding settings, but it is important that any changes made to
+    # master-conf-dir and master-code-dir are also made to the corresponding puppet
+    # settings when running the puppet CLI tools. See
+    # https://docs.puppetlabs.com/puppetserver/latest/puppet_conf_setting_diffs.html#overriding-puppet-settings-in-puppet-server
+    # for more information.
+
     # (optional) path to puppet conf dir; if not specified, will use the puppet default
```

Comment:
I think you should go ahead and add the actual defaults for codedir and condfir, like you did for the others. It's true that they use the Puppet defaults, but it's hardcoded in Puppet Server, and I think it's just more useful to say it explicitly.